core/backend/src/BlobContainerService.ts

```diff
@@ -0,0 +1,142 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module BlobContainers
+ */
+
+// spell:ignore datacenter
+
+import { AccessToken, Id64String } from "@itwin/core-bentley";
+
+/**
+ * Types and functions for creating, deleting and authorizing access to cloud-based blob containers for an iTwin.
+ * @beta
+ */
+export namespace BlobContainer {
+
+  /** Object that implements the methods to create, delete, and request access to a container. */
+  export let service: BlobContainer.ContainerService | undefined;
+
+  /** name of cloud provider for a container. */
+  export type Provider = "azure" | "google" | "aws";
+
+  /** the name of the container within its `Scope` */
+  export type ContainerId = string;
+
+  /** token that authenticates a user. This token is required to obtain a `ContainerToken`. */
+  export type UserToken = AccessToken;
+
+  /** token that authenticates access to a container for either read or write. */
+  export type ContainerToken = AccessToken;
+
+  /**
+   * The scope for a container. This determines:
+   *  - the "owner" organization, including the contract that governs its legal and commercial terms and obligations
+   *  - the administrators who may configure RBAC permissions
+   *  - the datacenter for the container. Every container resides in a datacenter as determined by the iTwinId. This
+   * determines the region for data residency requirements.
+   *  - the lifecycle constraints for the container. No container can outlive its iTwin or its iModel (if defined). That is,
+   * when the iTwin/iModel is deleted (either explicitly or due to contract expiration), the container is also deleted.
+   */
+  export interface Scope {
+    /** iTwinId of the owner of this container. */
+    iTwinId: Id64String;
+    /** optionally, an iModelId within the iTwin. If present, container is deleted when the iModel is deleted. */
+    iModelId?: Id64String;
+  }
+
+  /**
+   * Metadata about the use of a container so that:
+   *  - administrators can understand why a container exists for assigning RBAC permissions appropriately
+   *  - usage reports can aggregate types of containers
+   *  - applications can identify their containers
+   *  - applications can store important (to them) properties "on" their containers
+   */
+  export interface Metadata {
```

Comment:
Which of these properties would be used to serve the purpose of "usage reports can aggregate types of containers"?

Reply:
`format` and `application`, for example.

Reply:
These fields line up with what we are tracking currently. In addition, we also have a human readable, non-unique "name" for the container. In the case of "type", type was originally CloudSQLite specific for controlling what would be stored inside the CloudSQLite database, that would be replaced by format + application in this case?

Reply:
I don't see how format + application replaces 'type' unless we infer the type from the application, but I believe that would require us to assume that each application only creates one type. 

How about something like:
export type = 'workspace' | 'codes' | 'propertystore' (These types may be wrong) 

This would also loosely prevent services from creating containers for information that we don't wish to store. They'd have to ask the platform/blobcontainerservice whether we could support a new type or whether their use case fits in with the blobcontainerservice.

Reply:
Do we need both "name" and "description"? I suppose one can be optional. I don't think "name" is the right term though, since it isn't unique. 

Probably we need to think about which of these fields are defined by the application that creates them and which are meant to be free-form for users to recognize them for their own purposes. For example, there will be lots of containers for workspaces. They should have "format=workspace" (@nick4598 i think "format" and "type" are the same concept), so we know they're workspaces. Then, they'll probably have an application that creates them "application=substation". Then they'll have a "description=`templates for electrical components`". Maybe they also need something the user can supply, e.g. "notes=`approved for all Acme projects`". 

@MichaelSwigerAtBentley, don't we also need something like "RBACRole", so the RBAC control panel will know how to associate permissions with a given container. E.g. "rbacRole=project-workspace". Admittedly, i don't know much about your RBAC plan.

Reply:
When I read the comment above format: "a machine-readable string that describes the "format" of the data in this container (e.g. "CloudSqlite")", I interpret that to mean CloudSqlite would be one of the potential options for this property. That would be at odds with a format of workspace, right? Since I thought workspaces would be accessed or stored in CloudSqlite form, i.e. manifest and blocks?

Reply:
maybe i should change the comment to use "workspace" as the example. I think of this string playing the same role as "extension" in filenames.